client/transactions/list/index.tsx

```diff
@@ -684,16 +690,23 @@ export const TransactionsList = (
 					method: 'POST',
 				} );
 
-				createNotice(
-					'success',
-					sprintf(
-						__(
-							'Your export will be emailed to %s',
-							'woocommerce-payments'
-						),
-						userEmail
-					)
-				);
+				if ( response?.download_url ) {
+					const link = document.createElement( 'a' );
+					link.href = response.download_url;
+					link.click();
```

Comment:
💭 Note, this could be a `window.location.href`, but maybe there's a benefit to `link.click()` that I'm not thinking about?

```suggestion
					window.location.href = response.download_url;
```

Reply:
Won't `window.location.href` load the current file in that tab? I was trying to get the same behaviour that we have now with `saveFile` which is to download the file in background.

Reply:
Ah, you're right – in my current browser (Arc) the download is triggered when using `window.location.href`, but this behaviour would not be as predictable across browsers as `link.click()`.

As part of this PR review testing, we should check this `link.click()` download behaviour in various browsers, aiming for consistency.

Reply:
I've briefly tested this in the following browsers:

- Arc (Chromium) triggers a file download ✅
- Edge (Chromium) triggers a file download ✅
- Safari opens the CSV file in the current browser tab, doesn't trigger a download ⚠️
- Firefox triggers a file download ✅

To help Safari, adding the following attributes on the link helps by opening the file in a new tab. But this doesn't solve the CSV being shown in a browser tab rather than downloaded. 🤔

```ts
link.setAttribute( 'download', '' ); // I don't think this will work for Safari, since the target is at a different origin https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#:~:text=only%20works%20for%20same%2Dorigin%20URLs
link.setAttribute( 'target', '_blank' );
link.setAttribute( 'rel', 'noopener noreferrer' );
```

I experimented with downloading the CSV content as a blob and using the `filesaver` package `saveAs`  (use by the current implementation in `@woocommerce/csv-export` [source](https://github.com/woocommerce/woocommerce/blob/0b59a8986577bbb9e5cb7f68e1ce209874674c51/packages/js/csv-export/src/index.ts#L139-L144)), but this introduces CORS complexities since the CSV is generated and served at different origin to the browser tab (WPCOM).

```ts
const response = await fetch(downloadUrl);
const blob = await response.blob();
const url = window.URL.createObjectURL(blob);
const link = document.createElement('a');
link.href = url;
link.click();
URL.revokeObjectURL(url); // cleanup
```

Reply:
Thanks, we can also use filesaver with a URL, but it didn't seem to be working.

Reply:
I wonder if we can instruct browsers to download this rather than display, using the `Content-Disposition` header.

I don't see this header in the response from server, e.g. `/wp-content/uploads/2025/01/woocommerce_payments_export-acct_1pxrhsr4vt8oow8f-2025_01_08_02_00_28-ydyglge144.csv`.

```
Content-Disposition: attachment; filename="export_name.csv"
Content-Type: text/csv; charset=utf-8
```

[Content-Disposition - HTTP | MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition)

Reply:
> I wonder if we can instruct browsers to download this rather than display, using the Content-Disposition header.

Yes we can! And it totally solves the problem in Safari 🎉

To fix, add the following after the existing `update_post_meta` in `Export_Service::upload_file_to_wpcom`:

```php
update_post_meta( $attachment_id, '_wp_attachment_content_disposition', 'attachment; filename="' . sanitize_file_name( $filename ) . '"' );
```

This is probably better as a separate issue/PR, since it will impact any use of `Export_Service::upload_file_to_wpcom`, but related to this implementation. Or, it could be included in https://github.com/Automattic/transact-platform-server/pull/7077 if we think the changes are low-risk.

Reply:
> I wonder if we can instruct browsers to download this rather than display, using the Content-Disposition header.

Yes we can! And it totally solves the problem in Safari 🎉

~~To fix, add the following after the existing `update_post_meta` in `Export_Service::upload_file_to_wpcom`:~~

```php
update_post_meta( $attachment_id, '_wp_attachment_content_disposition', 'attachment; filename="' . sanitize_file_name( $filename ) . '"' );
```

🎉🎉 **Update** an even easier solution is to add `force_download=true` to the `download_url` which is the preferred solution for WPCOM-served files for adding the appropriate `Content-Disposition` headers – see pMz3w-fJo-p2:

```suggestion
					const link = document.createElement( 'a' );
					// Add force_download=true to the URL to force the download, which adds the appropriate `Content-Disposition: attachment` header when using production server.
					link.href = response.download_url + '?force_download=true';
					link.click();
```

Reply:
> Safari opens the CSV file in the current browser tab, doesn't trigger a download ⚠️

I've opened #7095 since this issue also exists when clicking "Download Export" in emails.